Modix.Data/Models/Promotions/PromotionCampaignSearchCriteria.cs

```diff
@@ -91,22 +91,22 @@ public static IQueryable<PromotionCampaignEntity> FilterBy(this IQueryable<Promo
                     x => x.TargetRoleId == criteria.TargetRoleId,
                     !(criteria.TargetRoleId is null))
                 .FilterBy(
-                    x => x.CreateAction.Created >= criteria.CreatedRange.Value.From.Value,
+                    x => x.CreateAction.Created >= criteria.CreatedRange!.Value.From!.Value,
                     !(criteria.CreatedRange?.From is null))
                 .FilterBy(
-                    x => x.CreateAction.Created <= criteria.CreatedRange.Value.To.Value,
+                    x => x.CreateAction.Created <= criteria.CreatedRange!.Value.To!.Value,
                     !(criteria.CreatedRange?.To is null))
                 .FilterBy(
                     x => x.CreateAction.CreatedById == criteria.CreatedById,
                     !(criteria.CreatedById is null))
                 .FilterBy(
-                    x => x.CloseAction.Created >= criteria.ClosedRange.Value.From.Value,
+                    x => x.CloseAction != null && x.CloseAction!.Created >= criteria.ClosedRange!.Value.From!.Value,
```

Comment:
This check and the one below it are the only real bugs I found in this PR (provided I annotated correctly). AFAICT specifying ClosedRange does not guarantee there was a CloseAction, so this was a potential null ref.

Reply:
I actually think I remember this specifically, and IIRC, it's not a potential NRE because this is an expression that only ever translates into SQL, where it's just comparing against a column on a table that's `LEFT JOIN`ed in. What this should generate into is something like...
```sql
WHERE CloseAction.Id IS NOT NULL AND CloseAction.Created >= @ClosedRangeFrom
```
...and the `IS NOT NULL` check is technically redundant.

Reply:
Alright, I'll revert the additional checks then.